core/services/network_checker.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require_relative 'log_storage'
+require_relative 'machine_configurator'
+
+# This class checks the health of network resources
+module NetworkChecker
+  RESOURCES = ['https://github.com',
```

Comment:
When creating an array of strings prefer `%w` style. This way you will not need to put all quotes into the text.